### ISSUE TITLE
Failed detecting return type change

### ISSUE THREAD

**Issue by lauraluiz** [revapi/revapi#56]
In our SDK we have recently introduced code auto-generation and we decided to use Revapi Maven Plugin to make sure we were not introducing breaking changes. So far worked great, but apparently the plugin wasn't able to find a return type change which raised the nasty `NoSuchMethodError` exception to our SDK users under certain circumstances.

I've tried to narrow down the info as much as possible, to be able to figure out where the problem comes from. Everything relevant can be found in https://gist.github.com/lauraluiz/1cb2f1ded27bebeae37b0d0e9dda01b6, where it is analyzed the SDK from versions v1.6.0 to v1.8.0, where the change happened (in case you wonder, there is a small anomaly on the v1.7.0 Javadoc, reason why I've decided to skip it).

In particular, the return type was changed from returning an interface to returning one of the implementations. The source code remained the same, the signature didn't provide any breaking change. But then of course the Java bytecode changed and couldn't find the method's signature anymore, thus raising the `NoSuchMethodError`.

The report doesn't show this change though. Is it expected, a bug or did I miss something?

**Comment by metlos** [comment]
Well, this is definitely a bug caused by me being simply stupid @ https://github.com/revapi/revapi/blob/master/revapi-java-spi/src/main/java/org/revapi/java/spi/Code.java#L148 😃 A covariant return type change definitely breaks binary compatibility.

Thanks for the report!

**Comment by metlos** [comment]
You can workaround this problem for time being by reclassifying the covariant return type change as a breaking change using:
```javascript
{
  "revapi": {
    "reclassify": [
      {
        "code": "java.method.returnTypeChangedCovariantly",
        "classify": {
          "BINARY": "BREAKING"
        }
      }   
    ]
  }
}
```

But a fix for this will be in `revapi-java-0.12.0`. I will also be updating the website to contain all of the found problems with slightly improved examples, etc.

**Comment by lauraluiz** [comment]
Great fast fix!
And thanks for the workaround :)
